components/camel-file/src/main/java/org/apache/camel/component/file/FileOperations.java

```diff
@@ -381,6 +381,9 @@ public boolean storeFile(String fileName, Exchange exchange, long size) throws G
                 // buffer the reader
                 in = IOHelper.buffered(in);
                 writeFileByReaderWithCharset(in, file, charset);
+            } else if (exchange.getIn().getBody() instanceof String &&
+                    ((String) exchange.getIn().getBody()).length() < endpoint.getBufferSize()) {
```

Comment:
Maybe put the body into a String str = (String) ... so you avoid the ugly type casts